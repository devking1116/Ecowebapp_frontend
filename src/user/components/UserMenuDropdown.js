/* @flow */
import React from 'react';
import { List, Popup } from 'semantic-ui-react';
import reactCSS from 'reactcss';

type Props = {
  icon: any,
  isOpen?: bool,
  children?: any,
};

/**
 * A dropdown for the user menu.
 */
export default function UserMenuDropdown({ icon, children, isOpen, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      icon: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        height: '100%',
      },
    },
  });

  return (
    <Popup
      trigger={<div style={styles.icon}>{icon}</div>}
      flowing
      hoverable
      {...otherProps}
    >
      <List divided relaxed>
      {React.Children.map(children, (child) =>
        <List.Item>
          {child}
        </List.Item>
      )}
      </List>
    </Popup>
  );
}
