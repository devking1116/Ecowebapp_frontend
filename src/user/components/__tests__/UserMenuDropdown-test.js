import React from 'react';
import renderer from 'react-test-renderer';

import UserMenuDropdown from '../UserMenuDropdown';

test('it displays the given icon.', () => {
  const component = renderer.create(
    <UserMenuDropdown icon={<div>Test Icon</div>} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

//NOTE, unable to perform this test because semantic-ui-react will not allow Popup to start open (???)
test.skip('it only displays the dropdown if isOpen is set.', () => {
  const closedDropdownComponent = renderer.create(
    <UserMenuDropdown>
      THIS SHOULD NOT BE DISPLAYED
    </UserMenuDropdown>
  );
  const openDropdownComponent = renderer.create(
    <UserMenuDropdown open>
      This should be displayed.
    </UserMenuDropdown>
  );

  const closedDropdownTree = closedDropdownComponent.toJSON();
  const openDropdownTree = openDropdownComponent.toJSON();

  expect(closedDropdownTree).toMatchSnapshot();
  expect(openDropdownTree).toMatchSnapshot();
});
