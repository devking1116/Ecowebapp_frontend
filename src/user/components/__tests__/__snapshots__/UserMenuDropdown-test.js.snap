exports[`test it displays the given icon. 1`] = `
<div
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "WebkitJustifyContent": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
    }
  }>
  <div>
    Test Icon
  </div>
</div>
`;
